common/util/slices.go

```diff
@@ -14,3 +14,11 @@ func UniqueSliceStringer[T fmt.Stringer](s []T) []T {
 	}
 	return result
 }
+
+func SliceToSetStringer[T fmt.Stringer](s []T) map[string]struct{} {
```

Comment:
This can be removed if we iterate over the slice instead of the map (see other comment).